starknet_py/net/signer/test_ledger_signer.py

```diff
@@ -0,0 +1,173 @@
+            constructor_calldata=[1, 2, 3, 4],
+            max_fee=10000,
+            signature=[],
+            nonce=23,
+            version=1,
+        ),
+        DeclareV1(
+            contract_class=create_compiled_contract(
+                compiled_contract=compiled_contract
+            ),
+            sender_address=123,
+            max_fee=10000,
+            signature=[],
+            nonce=23,
+            version=1,
+        ),
+    ],
+)
+def test_sign_transaction(transaction):
+    signer = LedgerSigner(
+        derivation_path_str="m/2645'/1195502025'/1470455285'/0'/0'/0",
+        chain_id=StarknetChainId.MAINNET,
+    )
+
+    signature = signer.sign_transaction(transaction)
+
+    assert isinstance(signature, list)
+    assert len(signature) > 0
+    assert all(isinstance(i, int) for i in signature)
+    assert all(i != 0 for i in signature)
+
+
+async def _get_account_balance_eth(client: FullNodeClient, address: int):
+    eth_address = int(
+        "0x49D36570D4E46F48E99674BD3FCC84644DDD6B96F7C741B1562B82F9E004DC7", 16
+    )
+    balance = await client.call_contract(
+        call=Call(
+            to_addr=eth_address,
+            calldata=[address],
+            selector=get_selector_from_name("balanceOf"),
+        )
+    )
+    return balance
+
+
+@pytest.mark.asyncio
+async def test_deploy_account_and_transfer(client):
+    signer = LedgerSigner(
+        derivation_path_str="m/2645'/1195502025'/1470455285'/0'/0'/0",
+        chain_id=StarknetChainId.SEPOLIA,
+    )
+
+    class_hash = int(
+        "0x61dac032f228abef9c6626f995015233097ae253a7f72d68552db02f2971b8f", 16
+    )
+
+    salt = 1
+    calldata = [signer.public_key]
+    address = compute_address(
+        salt=salt,
+        class_hash=class_hash,
+        constructor_calldata=calldata,
+    )
+    account = Account(
+        address=address,
+        client=client,
+        signer=signer,
+        chain=StarknetChainId.SEPOLIA,
+    )
+    await mint_token_on_devnet(
+        url=client.url.replace("/rpc", ""),
+        address=address,
+        amount=5000000000000000000000,
+        unit="WEI",
+    )
+
+    signed_tx = await account.sign_deploy_account_v1(
+        class_hash=class_hash,
+        contract_address_salt=salt,
+        constructor_calldata=calldata,
+        auto_estimate=True,
+    )
+
+    await client.deploy_account(signed_tx)
+
+    recipient_address = int(
+        "0x1323cacbc02b4aaed9bb6b24d121fb712d8946376040990f2f2fa0dcf17bb5b", 16
+    )
+    sender_balance_before = (await _get_account_balance_eth(client, address))[0]
+    recipient_balance_before = (
+        await _get_account_balance_eth(client, recipient_address)
+    )[0]
+    print(f"Balance of {address} before: {sender_balance_before}")
+    print(f"Balance of {recipient_address} before: {recipient_balance_before}")
+
+    eth_address = int(
+        "0x49D36570D4E46F48E99674BD3FCC84644DDD6B96F7C741B1562B82F9E004DC7", 16
+    )
+
+    call = Call(
```

Comment:
Maybe it would be better to use `Contract` class for interactions with the ETH contract?

Reply:
Sure, updated.

Reply:
That's how the docs look now
<img width="592" alt="image" src="https://github.com/user-attachments/assets/71744823-9e77-4de2-af12-265cbf1f67d0">
<img width="566" alt="image" src="https://github.com/user-attachments/assets/b0f335a3-b74c-4a4a-8727-933347dd3844">
Unfortunately, we cannot get rid of the prefund-related code in the second section. Maybe we can just start the example after prefunding, wdyt @ddoktorski ?